src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -329,6 +349,12 @@ export const useSearch = (services: DiscoverViewServices) => {
         }
       }
 
+      // Currently error message is sent as encoded JSON string, which requires extra parsing
+      // TODO: Confirm error contract
+      if (isJSONString(errorBody.message)) {
+        errorBody.message = JSON.parse(errorBody.message);
```

Comment:
nit. maybe define something like `safeParseJSONString` to avoid calling JSON.parse twice?